cmd/gardenlet/app/gardenlet.go

```diff
@@ -204,18 +205,19 @@ These so-called control plane components are hosted in Kubernetes clusters thems
 // Gardenlet represents all the parameters required to start the
 // Gardenlet.
 type Gardenlet struct {
-	Config                 *config.GardenletConfiguration
-	Identity               *gardencorev1beta1.Gardener
-	GardenNamespace        string
-	GardenClusterIdentity  string
-	ClientMap              clientmap.ClientMap
-	K8sGardenCoreInformers gardencoreinformers.SharedInformerFactory
-	KubeInformerFactory    informers.SharedInformerFactory
-	Logger                 *logrus.Logger
-	Recorder               record.EventRecorder
-	LeaderElection         *leaderelection.LeaderElectionConfig
-	HealthManager          healthz.Manager
-	CertificateManager     *certificate.Manager
+	Config                     *config.GardenletConfiguration
+	Identity                   *gardencorev1beta1.Gardener
+	GardenNamespace            string
+	GardenClusterIdentity      string
+	ClientMap                  clientmap.ClientMap
+	K8sGardenCoreInformers     gardencoreinformers.SharedInformerFactory
+	K8sSeedManagementInformers seedmanagementinformers.SharedInformerFactory
```

Comment:
OK, I will try, though I am not sure I fully understand all the implications. I will reach back if I have doubts.

Reply:
@timebertt is there an existing controller that can be taken by Stoyan as reference?

Reply:
@stoyanr I changed my mind on this. I underestimated the implications of using c-r clients/informers here in this new controller.
This will get a bit messy, because it will require a duplicated watch+cache for Shoots (even if the `CachedRuntimeClients` feature gate is disabled), which we should try to avoid. Also, you probably won't be able to use some of the filtering functions and so on with the c-r informers.
So please continue with the generated clientset and informers like you did before and let me do the switch to c-r informers in one run with the other controllers, when I have the full picture and overview. I don't want to block your PR with this too much.
Sorry for the back and forth.